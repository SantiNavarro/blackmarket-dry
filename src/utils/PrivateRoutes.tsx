import { Outlet, Navigate } from 'react-router-dom';
import { selectUserData } from '../store/features/api/selectors';
import { useAppSelector } from '../store/hooks';

const PrivateRoutes = () => {
  const userState = useAppSelector(selectUserData);

  return userState.email ? <Outlet /> : <Navigate to="/sign-in" />;
};

export default PrivateRoutes;
