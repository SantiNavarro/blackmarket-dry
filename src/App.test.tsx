import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { QueryClient, QueryClientProvider } from 'react-query';

import { MemoryRouter, Routes, Route } from 'react-router-dom';
import PrivateRoutes from './utils/PrivateRoutes';
import HomePage from './pages/Home';
import SignInPage from './pages/Login';
import SignUpPage from './pages/SignUp';
import ProductsPage from './pages/Products';
import userReducer from './store/slices/userSlice';

describe('My app', () => {
  it('renders correctly', () => {
    const store = configureStore({
      reducer: {
        user: userReducer,
      },
    });
    const queryClient = new QueryClient();

    const renderer = create(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <MemoryRouter initialEntries={['/sign-in']}>
            <Routes>
              <Route element={<PrivateRoutes />}>
                <Route element={<HomePage />} path="/" />
                <Route element={<ProductsPage />} path="/products" />
              </Route>
              <Route element={<SignInPage />} path="/sign-in" />
              <Route element={<SignUpPage />} path="/sign-up" />
            </Routes>
          </MemoryRouter>
        </QueryClientProvider>
      </Provider>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
