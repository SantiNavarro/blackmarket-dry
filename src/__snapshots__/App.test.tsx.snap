// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My app renders correctly 1`] = `
<div
  className="login"
>
  <div
    className="basic-form-position"
  >
    <div
      className="basic-form"
    >
      <div
        className="basic-form__logo"
        role="img"
      />
      <div
        className="form-control-container"
      >
        <div
          className="inputs-container"
        >
          <div
            className="MuiFormControl-root css-1uitn3s-MuiFormControl-root"
          >
            <p>
              Email
            </p>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl outlined-adornment-email css-1av2g8m-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="outlined-adornment-email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Type your email"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-hdw1oc"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root css-1uitn3s-MuiFormControl-root"
          >
            <p
              className="paragraph"
            >
              Password
            </p>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1sxxfr2-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
              required={true}
            >
              <input
                aria-invalid={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="outlined-adornment-password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Type your password"
                required={true}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-hdw1oc"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          className="submit-form-button-disabled"
          disabled={true}
          onClick={[Function]}
          onSubmit={[Function]}
          type="button"
        >
          Log in
        </button>
        <p
          className="paragraph-secondary"
        >
          I forgot my password
        </p>
      </div>
    </div>
    <div
      className="basic-form"
    >
      <div
        className="basic-form"
      >
        <div
          className="dontHaveAccount"
        >
          <p>
            Dont have an account?
          </p>
          <button
            className="submit-form-button-secondary"
            onClick={[Function]}
            type="button"
          >
            Sign up
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
